### ISSUE TITLE
$GOPATH is unset

### ISSUE THREAD

**Issue by asdwsda** [CircleCI-Public/cimg-go#25]
`$GOPATH` is unset during image build, and because of that, `$PATH` would be wrong as well.

Expected (let's assume that `GOPATH='/go'`)
```
$ echo $PATH
/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

Actual:
```
$ echo $PATH
/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```
Notice that `/bin` is duplicated.

**Comment by felicianotech** [comment]
Looking into this.

**Comment by felicianotech** [comment]
@asdwsda It's looking to me that `GOPATH` is deprecated as of Go v1.13.

The biggest use I still see for it is when `go get` installing binaries, which not only should be used less and less, but shouldn't be a CI activity in my opinion.

Can you explain your use-case here?

**Comment by asdwsda** [comment]
You are right, `GOPATH` is deprecated in Go v1.13. However this repo contains v1.12 and v1.11 as well. What about these versions?

I agree with you. Installing binaries shouldn't be a CI activity, but sometimes I need to do it. I think it is expected to be able to use `go get` and installed bins in such containers.

What do you think about setting [GOBIN](https://golang.org/cmd/go/#hdr-Environment_variables) instead of `GOPATH`?

> GOBIN
> 	The directory where 'go install' will install a command.

**Comment by felicianotech** [comment]
`GOBIN` might be viable. Looking into this and will decide on something soon.

**Comment by felicianotech** [comment]
@asdwsda I did some testing.

1. go get will assume a `GOPATH` for you when installing a binary if one isn't set. That being said, the resulting location still wasn't in `PATH` so not really useful.
2. `GOBIN` was a good idea and would work. While doing some testing, it felt like a half-measure to just properly fixing `GOPATH` so I just did that instead.
3. PR #29 will make `GOPATH` and `PATH` work for Go binaries.
